tests/macsec/test_dataplane.py

```diff
@@ -198,3 +186,52 @@ def test_counters(self, duthost, ctrl_links, upstream_links, rekey_period, wait_
                 assert ingress_end_counters[i] - ingress_start_counters[i] >= PKT_NUM * PKT_OCTET
             else:
                 assert ingress_end_counters[i] - ingress_start_counters[i] >= PKT_NUM
+
+    def test_clear_counters(self, duthost, ctrl_links, upstream_links, rekey_period, wait_mka_establish):
+        if rekey_period:
+            pytest.skip("Counter increase is not guaranteed in case rekey is happening")
+
+        PKT_NUM = 5
+        PKT_OCTET = 1024
+
+        # Select some one macsec link
+        port_name = list(ctrl_links)[0]
+        nbr_ip_addr = upstream_links[port_name]['local_ipv4_addr']
+        pc = find_portchannel_from_member(port_name, get_portchannel(duthost))
+        if pc:
+            assert pc["status"] == "Up"
+            up_ports = pc["members"]
+        else:
+            up_ports = [port_name]
+
+        # Launch some traffic
+        ret = duthost.command(
+            "{} ping -c {} -s {} {}".format(get_ipnetns_prefix(duthost, port_name), PKT_NUM, PKT_OCTET, nbr_ip_addr))
+        assert not ret['failed']
+        sleep(10)   # wait 10s for polling counters
+
+        egress_end_counters, ingress_end_counters = self._get_counters(duthost, up_ports)
+
+        clear_macsec_counters(duthost)
+
+        egress_cleared_counters, ingress_cleared_counters = self._get_counters(duthost, up_ports)
+
+        for counter in egress_end_counters:
+            if counter.startswith('SAI_MACSEC_SA_STAT'):
+                if egress_end_counters[counter] != 0:
+                    assert egress_cleared_counters[counter] < egress_end_counters[counter]
+            else:
+                # Non stats should not reset
+                # TODO: Currently XPN is reset - uncomment this when fixed
```

Comment:
this needs this PR to be enabled : https://github.com/sonic-net/sonic-buildimage/pull/20654